src/Middleware/HttpLogging/src/HttpLoggingContext.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpLogging;
+
+/// <summary>
+/// The context used for logging customization callbacks.
+/// </summary>
+/// <param name="httpContext">The request context.</param>
+public sealed class HttpLoggingContext(HttpContext httpContext)
```

Comment:
While you are looking at this, make sure the tries to minimize allocations (not allocating this if it's not needed and looking at if this can be a struct vs class etc).

Reply:
I'd added a check up front to skip if there are no interceptors and no fields enabled by default or attribute. 

Other than that, I think our best bet is to cache these instances and their parameter lists.

Reply:
Only allocating if HttpLogging is enabled and there are logging interceptors configured seems good to me. One extra pay-for-play allocation across the request is acceptable.